nomad/structs/extensions.go

```diff
@@ -8,8 +8,10 @@ var (
 	// extendedTypes is a mapping of extended types to their extension function
 	// TODO: the duplicates could be simplified by looking up the base type in the case of a pointer type in ConvertExt
```

Comment:
totally untested (and unnecessary), but you might also be able to re-frame this map as a type switch with the 2 types per case statement

Reply:
It looks like we could that in [`ConvertExt`](https://github.com/hashicorp/nomad/blob/main/nomad/structs/encoding.go#L17) but we'd still need the list for calling `SetInterfaceExt` on the codec in [`NomadJsonEncodingExtensions `](https://github.com/hashicorp/nomad/blob/main/nomad/structs/encoding.go#L33)